### PR TITLE
Add leaf snowflake key generator and related functions.

### PR DESCRIPTION
**Pull request by wgy8283335**
Changes proposed in this pull request:

Add LeafSnowflakeKeyGenerator class and TimeService class.
Add LeafSnowflakeKeyGeneratorTest class and FixedTimeService class.

### CODE REVIEW

**Comment by wgy8283335** [comment]
@tristaZero.During CI building, the error happened with message "The command "travis_wait 30 mvn install --quiet" failed and exited with 137 during ." Is there anything wrong in the Travis CI system?

**Comment by sunbufu** [comment]
```bash
Still running (30 of 30): mvn install --quiet
Timeout (30 minutes) reached. Terminating "mvn install --quiet"
```
It looks like CI check timed out. I think you should try it again.

File: sharding-orchestration/sharding-orchestration-core/src/test/java/org/apache/shardingsphere/orchestration/internal/keygen/LeafSnowflakeKeyGeneratorTest.java [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+        properties.setProperty("serverList", "127.0.0.1:2181");
+        properties.setProperty("serviceId", "testService1");
+        properties.setProperty("maxTimeDifference", "5000");
+        properties.setProperty("registryCenterType", "ForthTestRegistryCenter");
+        leafSnowflakeKeyGenerator.setProperties(properties);
+        Set<Comparable<?>> actual = new HashSet<>();
+        int taskNumber = threadNumber << 2;
+        for (int i = 0; i < taskNumber; i++) {
+            actual.add(executor.submit(new Callable<Comparable<?>>() {
+                @Override
```
Comment (tristaZero) [inline sharding-orchestration/sharding-orchestration-core/src/test/java/org/apache/shardingsphere/orchestration/internal/keygen/LeafSnowflakeKeyGeneratorTest.java]: Add blank line

File: sharding-orchestration/sharding-orchestration-core/src/test/java/org/apache/shardingsphere/orchestration/internal/keygen/LeafSegmentKeyGeneratorTest.java [hunk 2]
```diff
@@ -142,12 +138,11 @@ public void assertGenerateKeyWithDefaultStep() {
                 }
             }).get());
         }
-        assertThat(actual.size(), is(taskNumber));
+        assertThat(actual.size(), CoreMatchers.is(taskNumber));
```
Comment (tristaZero) [inline sharding-orchestration/sharding-orchestration-core/src/test/java/org/apache/shardingsphere/orchestration/internal/keygen/LeafSegmentKeyGeneratorTest.java]: Please use import static if you want to use is() for asserting.

**Pull request merged**